### ISSUE TITLE
SelectMany over primitive collection property fails

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#32505]
Query:

```c#
_ = await ctx.Blogs.SelectMany(b => b.Tags).ToListAsync();
```

<details>
<summary>Full repro</summary>

```c#
await using var ctx = new BlogContext();
await ctx.Database.EnsureDeletedAsync();
await ctx.Database.EnsureCreatedAsync();

ctx.Blogs.AddRange(
    new Blog { Tags = new List<string> { "tag1", "tag2" } },
    new Blog { Tags = new List<string> { "tag2", "tag3" } });

_ = await ctx.Blogs.SelectMany(b => b.Tags).ToListAsync();

public class BlogContext : DbContext
{
    public DbSet<Blog> Blogs { get; set; }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseSqlServer("Server=localhost;Database=test;User=SA;Password=Abcd5678;Connect Timeout=60;ConnectRetryCount=0;Encrypt=false")
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();
}

public class Blog
{
    public int Id { get; set; }
    public List<string> Tags { get; set; }
}
```

</details>

The cause of the issue is that the SelectMany lambda is visited by the top-level QueryableMethodTranslatingExpressionVisitor - not by a subquery visitor. Our code for identifying the property access (b.Tags) kicks in only after the regular translation failed ([code](https://github.com/dotnet/efcore/blob/main/src/EFCore.Relational/Query/RelationalQueryableMethodTranslatingExpressionVisitor.cs#L268)); but the base visitation throws before that when translation fails on a non-subquery visitor ([code](https://github.com/dotnet/efcore/blob/main/src/EFCore/Query/QueryableMethodTranslatingExpressionVisitor.cs#L520)).

**Comment by roji** [comment]
@mu88 definitely not on purpose; if you're seeing a SQLite problem here, please open a new issue with a minimal repro and we'll investigate.

**Comment by roji** [comment]
Ah, that's quite different indeed, and unrelated to this issue (#32505). SQLite itself does not support CROSS APPLY (or LATERAL JOIN), and so it's not possible to translate various query shapes to SQLite which work in other databases. Usage of SelectMany where the outer isn't referenced should work (as they don't require CROSS APPLY/LATERAL JOIN).

This bug here was about internal EF support for translating SelectMany (regardless of databases), whereas you're now running into a SQLite limitation.

**Comment by mu88** [comment]
Okay, thanks for the clarification 👍🏻 I suggest you update my original issue (#34479) with a similar comment so that a future reader (maybe one of us 😉) doesn't get irritated, too.

**Comment by roji** [comment]
@mu88 feel free to post a comment there linking here or similar.
